### PR TITLE
treewide: migrate slack-api-token for use with k8s creds plugin

### PR DESCRIPTION
**Pull request by dustymabe**
~~Also add `trySlackSend()` function, use it across the tree~~

### CODE REVIEW

File: utils.groovy [hunk 1]
```diff
@@ -194,6 +194,18 @@ def tryWithMessagingCredentials(Closure body) {
     }
 }
 
+// Send slack message if the credential exists. Do nothing if not.
+def trySlackSend(color, message) {
```
Comment (jlebon) [inline utils.groovy:198]: What does this get us? All the Slack sends are gated on `official` already. Keying off of the credential existing makes sense, but as a complexity trade I'd remove the `official` gates in that case.

Comment (jlebon) [reply utils.groovy:198]: Maybe simpler we skip that commit for now and circle back to it?

Comment (dustymabe) [reply utils.groovy:198]: yeah. The desire is to remove `official` in the future. I can drop that commit.

Comment (dustymabe) [reply utils.groovy]: removed that 2nd commit

**Review by jlebon — APPROVED** [review]

**Pull request merged**